### PR TITLE
Integrate option to pass clickThrough urls to renderAd method

### PR DESCRIPTION
**Pull request by renebaudisch**
## Type of change
- [x] Feature

## Description of change
This change enables publishers/marketers to count clicks on prebid inventory in their distributing adserver.
The publisher may call render ad with a third param accepting an object where right now only clickThrough will be checked but can be extended.
If this property is set we will replace any occurrence of ${CLICKTHROUGH} inside the properties ad and adUrl with the transmitted option.
If no occurrence is found nothing will be replaced.

## Other information
To make use of this feature bidAdpaters just have to respond with tags including this macro to pass it to their param and system

### CODE REVIEW

**Comment by patmmccann** [comment]
Chatting in the js committee today, we're hoping #5760 gets merged first, as renderers on display to track clicks might intefere with multiformat ads that need renderer on outstream.

**Comment by patmmccann** [comment]
Also, needs tests, thanks for the contribution!

File: src/prebid.js [hunk 1]
```diff
@@ -354,6 +354,12 @@ $$PREBID_GLOBAL$$.renderAd = function (doc, id) {
         // replace macros according to openRTB with price paid = bid.cpm
         bid.ad = utils.replaceAuctionPrice(bid.ad, bid.cpm);
         bid.adUrl = utils.replaceAuctionPrice(bid.adUrl, bid.cpm);
+
+        // replacing clickthrough if submitted
+        const { clickThrough } = options;
```
Comment (harpere) [inline src/prebid.js:359]: should validate that `options` is defined.

File: src/utils.js [hunk 2]
```diff
@@ -718,6 +718,11 @@ export function replaceAuctionPrice(str, cpm) {
   return str.replace(/\$\{AUCTION_PRICE\}/g, cpm);
 }
 
+export function replaceClickThrough(str, clicktag) {
+  if (!str) return;
+  return str.replace(/\$\{CLICKTHROUGH\}/g, clicktag);
```
Comment (harpere) [inline src/utils.js:723]: should validate that `clicktag` is defined and not NULL, maybe that it's a string.

**Review by harpere — CHANGES_REQUESTED** [review]
needs some validation and unit tests

**Review by harpere — CHANGES_REQUESTED** [review]
updates look good, thx, but still needs unit tests

**Comment by renebaudisch** [comment]
> updates look good, thx, but still needs unit tests

Ahm, yeah, tried my best... Is my first one ever, hope it fits.

**Comment by renebaudisch** [comment]
Closed #5816 in favor to #5820

**Review by harpere — APPROVED** [review]

**Review by jsnellbaker — APPROVED** [review]

**Comment by bretg** [comment]
I missed this one -- still kinda unclear on the real-world use case. The Prebid Universal Creative doesn't support this. 

@renebaudisch - could you outline a sample usage scenario so we can document this somewhere?

**Comment by renebaudisch** [comment]
When using an adserver to render or deliver the prebid response there is actually no option to count clicks by the delivering adserver by passing their clickThrough as one normally would do when adding a redirect for an IO booking.

By adding this options we could provide an clickthrough url from the adserver to the renderAd method so the adserver of the publisher would also count clicks on prebid creatives as we cannot right now.

This should also affect some reports and statistics of some adservers as the average clickrate right now has either to avoid calculate them or using their AIs by a clickrate of 0,00% what actually reduces this rate.

Sure, both sides have to support this, so the adCode response has to use a macro and the publisher has to provide the clickthrough but right now I just couldn't and this just gives us this possibility.
But if one of them won't nothing will change either for noone...

**Pull request merged**